packages/shell-api/src/shard.ts

```diff
@@ -205,21 +205,21 @@ export default class Shard<
     }
   }
 
-  async _getConfigDB(): Promise<DatabaseWithSchema<M, M[keyof M]>> {
+  async _getConfigDB(): Promise<DatabaseWithSchema<M, M['config']>> {
```

Comment:
I went back and forth on exactly this change and I seemed to just trade one set of errors for another. Although I agree that M['config'] is more correct - it is what I tried first.